cmake/CMakeLists.txt

```diff
@@ -1329,67 +1329,103 @@ if (onnxruntime_USE_DNNL)
+  set_target_properties(tvm_runtime PROPERTIES FOLDER ${tvm_SOURCE_DIR})
 
-    set(TVM_INCLUDES ${tvm_SOURCE_DIR}/include
-      ${tvm_SOURCE_DIR}/3rdparty/dmlc-core/include
-      ${tvm_SOURCE_DIR}/3rdparty/dlpack/include
-      $<TARGET_PROPERTY:tvm,INTERFACE_INCLUDE_DIRECTORIES>
-      $<TARGET_PROPERTY:tvm_topi,INTERFACE_INCLUDE_DIRECTORIES>)
-  endif()
+  set(TVM_INCLUDES ${tvm_SOURCE_DIR}/include
+    ${tvm_SOURCE_DIR}/3rdparty/dmlc-core/include
+    ${tvm_SOURCE_DIR}/3rdparty/dlpack/include
+    $<TARGET_PROPERTY:tvm,INTERFACE_INCLUDE_DIRECTORIES>
+    $<TARGET_PROPERTY:tvm_topi,INTERFACE_INCLUDE_DIRECTORIES>)
 
-  if (onnxruntime_USE_STVM)
-    set_target_properties(tvm PROPERTIES FOLDER ${tvm_SOURCE_DIR})
+  add_definitions(-DUSE_NUPHAR_TVM)
 
-    set(TVM_INCLUDES ${tvm_SOURCE_DIR}/include
-      ${tvm_SOURCE_DIR}/3rdparty/dmlc-core/include
-      ${tvm_SOURCE_DIR}/3rdparty/dlpack/include
-      $<TARGET_PROPERTY:tvm,INTERFACE_INCLUDE_DIRECTORIES>)
+  set(onnxruntime_tvm_libs onnxruntime_codegen_tvm)
+
+  # needs to link with stdc++fs in Linux
+  if (UNIX)
+    if (NOT APPLE)
+      set(FS_STDLIB stdc++fs)
+    endif()
   endif()
+  list(APPEND onnxruntime_EXTERNAL_LIBRARIES tvm ${FS_STDLIB})
+  list(APPEND onnxruntime_EXTERNAL_DEPENDENCIES tvm)
 
-  add_definitions(-DUSE_TVM)
+  if (WIN32 AND MSVC)
+    # wd4100: identifier' : unreferenced formal parameter
+    # wd4244: conversion from 'int' to 'char', possible loss of data
+    # wd4251: class X needs to have dll-interface to be used by clients of class Y
+    # wd4267: 'initializing': conversion from 'size_t' to 'int', possible loss of data
+    # wd4275: non dll-interface class X used as base for dll-interface class Y
+    # wd4389: signed/unsigned mismatch
+    # wd4456: declaration of X hides previous local declaration
+    set(DISABLED_WARNINGS_FOR_TVM "/wd4100" "/wd4244" "/wd4251" "/wd4267" "/wd4275" "/wd4389" "/wd4456")
+  else()
+    set(DISABLED_WARNINGS_FOR_TVM "-Wno-error=ignored-qualifiers" "-Wno-unused-parameter" "-Wno-ignored-qualifiers")
+    list(APPEND DISABLED_WARNINGS_FOR_TVM "-Wno-error=unused-parameter")
+    if (HAS_CATCH_VALUE)
+      #TODO: send a PR to TVM and fix it
+      list(APPEND DISABLED_WARNINGS_FOR_TVM "-Wno-error=catch-value")
+    endif()
+  endif()
+
+  include(onnxruntime_codegen.cmake)
+endif(onnxruntime_USE_NUPHAR_TVM)
+
+# TVM EP
+if (onnxruntime_USE_TVM)
+  if (NOT TARGET tvm)
+    message(STATUS "Include TVM(*).")
+    include(tvm)
+  endif()
 
-  if (onnxruntime_USE_NUPHAR)
-    set(onnxruntime_tvm_libs onnxruntime_codegen_tvm)
+  if (onnxruntime_USE_LLVM)
+    set(USE_LLVM ON CACHE BOOL "Only defined for TVM")
   endif()
-  if (onnxruntime_USE_STVM)
-    set(onnxruntime_tvm_libs onnxruntime_providers_stvm)
+
+  if (onnxruntime_TVM_CUDA_RUNTIME)
+    set(USE_CUDA ON CACHE BOOL "Only defined for TVM" FORCE)
   endif()
 
+  set(USE_OPENMP gnu CACHE STRING "Only defined for TVM")
+  # TODO(vvchernov): check messages and remove if need
```

Comment:
I added this messages to check cmake was doing was I was expected. Feel free to remove them.

Reply:
Ok, thanks! @xadupre  Can I remove it without CI test skipping?

Reply:
I meant the following messages. If you remove set(use_openmp..., it will be set for every package built after this one. That's why I did it that way. But I don't know the best setting. It could be set also with the same value as onnxruntime_USE_OPENMP. If it is better without, then let's remove it.